reana_db/models.py

```diff
@@ -627,6 +633,42 @@ def get_workspace_disk_usage(self, summarize=False, search=None):
 
         return [{"name": "", "size": disk_usage}]
 
+    def set_workspace_retention_rules(self, rules: List[Dict[str, str]]):
+        """Set workspace retention rules for the workflow."""
+        from .database import Session
+
+        for rule in rules:
```

Comment:
What do you think about having a limit on the number of rules? (in any case, we can do this in another PR)

Reply:
I think it could be tackled as part of [validation issue](https://github.com/reanahub/reana-server/issues/507). I've linked this comment there in the description